storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/ConsumerTaskTest.java

```diff
@@ -308,63 +294,62 @@ public void testCanReprocessSkippedRecords() throws InterruptedException {
-        thread.start();
+        consumerTask.ingestRecords();
 
-        TestUtils.waitForCondition(() -> consumerTask.isUserPartitionAssigned(tpId), "Waiting for " + tpId + " to be assigned");
         assertTrue(consumerTask.isMetadataPartitionAssigned(metadataPartition));
         assertFalse(handler.isPartitionInitialized.containsKey(tpId));
         IntStream.range(0, 5).forEach(offset -> addRecord(consumer, metadataPartition, tpId, offset));
-        TestUtils.waitForCondition(() -> consumerTask.readOffsetForMetadataPartition(metadataPartition).equals(Optional.of(4L)), "Couldn't read record");
+        consumerTask.ingestRecords();
+        assertEquals(Optional.of(4L), consumerTask.readOffsetForMetadataPartition(metadataPartition));
         assertTrue(handler.isPartitionInitialized.get(tpId));
     }
 
     @ParameterizedTest
     @CsvSource(value = {"0, 0", "500, 500"})
-    public void testMaybeMarkUserPartitionAsReadyWhenTopicIsEmpty(long beginOffset,
-                                                                  long endOffset) throws InterruptedException {
+    public void testMaybeMarkUserPartitionAsReadyWhenTopicIsEmpty(long beginOffset, long endOffset) {
         final TopicIdPartition tpId = getIdPartitions("world", 1).get(0);
         final int metadataPartition = partitioner.metadataPartition(tpId);
         consumer.updateBeginningOffsets(Collections.singletonMap(toRemoteLogPartition(metadataPartition), beginOffset));
         consumer.updateEndOffsets(Collections.singletonMap(toRemoteLogPartition(metadataPartition), endOffset));
         consumerTask.addAssignmentsForPartitions(Collections.singleton(tpId));
-        thread.start();
+        consumerTask.ingestRecords();
 
-        TestUtils.waitForCondition(() -> consumerTask.isUserPartitionAssigned(tpId), "Waiting for " + tpId + " to be assigned");
         assertTrue(consumerTask.isMetadataPartitionAssigned(metadataPartition));
-        TestUtils.waitForCondition(() -> handler.isPartitionInitialized.containsKey(tpId),
-            "should have initialized the partition");
+        assertTrue(handler.isPartitionInitialized.containsKey(tpId), "Should have initialized the partition");
         assertFalse(consumerTask.readOffsetForMetadataPartition(metadataPartition).isPresent());
     }
 
     @Test
     public void testConcurrentAccess() throws InterruptedException {
-        thread.start();
         final CountDownLatch latch = new CountDownLatch(1);
         final TopicIdPartition tpId = getIdPartitions("concurrent", 1).get(0);
         consumer.updateEndOffsets(Collections.singletonMap(toRemoteLogPartition(partitioner.metadataPartition(tpId)), 0L));
         final Thread assignmentThread = new Thread(() -> {
             try {
                 latch.await();
                 consumerTask.addAssignmentsForPartitions(Collections.singleton(tpId));
+                consumerTask.ingestRecords();
```

Comment:
I think this test might need an additional Thread. Since ingestRecords is only called on one thread normally, it's probably not a good idea to call it on two separate threads here. And I don't think the test has the same meaning as the original test if only one of assignmentThread and closeThread have the call to ingestRecords.